react-native-reanimated.d.ts

```diff
@@ -380,8 +413,56 @@ declare module 'react-native-reanimated' {
       initialValue: T
     ): AnimatedValue<T>;
 
+    // reanimated2 hooks
+    export function useSharedValue<T extends Value>(
+      initialValue: T
+    ): AnimatedValue<T>;
+    export function useDerivedValue<T extends Value>(
+      processor: () => T
+    ): AnimatedValue<T>;
+    export function useAnimatedStyle<T extends FlexStyle, TransformsStyle = ViewStyle>(
```

Comment:
You can use any styles you want here, not just View's styles

Reply:
Yeah, that’s why `T` extends `FlexStyle` and `TransformsStyle`. The default would be `ViewStyle` if generic isn’t passed. Check react-native types:
- [TextStyle](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cbb32bd37e8c941bd802c9eff19a845b7fe65043/types/react-native/index.d.ts#L784)
- [ImageStyle](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cbb32bd37e8c941bd802c9eff19a845b7fe65043/types/react-native/index.d.ts#L3531)